src/megaapi_impl.cpp

```diff
@@ -23166,96 +23166,118 @@ void MegaApiImpl::sendPendingRequests()
-                bool gotName = deviceName == name;
+                if(deviceNameNode->type != FOLDERNODE) { e = API_EACCESS; break; }
 
                 // get 'device-id' tag
-                const char* id = child->getCustomAttr("device-id");
-                bool gotId = deviceId == id;
+                const auto& childAttrs = deviceNameNode->attrs.map;
+                nameid attrId = AttrMap::string2nameid("dev-id"); // "device-id" would be too long
+                auto found = childAttrs.find(attrId);
 
-                // determine whether we found our node or not
-                if (!gotName && !gotId) { continue; } // ignore this child folder
+                // a folder without the attribute, or with it but with different value, will not be accessed
+                if (found == childAttrs.end() || found->second != deviceId) { e = API_EACCESS; break; }
+            }
+            else // create `DEVICE_NAME` remote dir
+            {
+                vector<NewNode> newnodes(1);
+                NewNode *newnode = &newnodes[0];
 
-                if (gotName && gotId) // got it, use this one to add another backup to it
-                {
-                    deviceNameNode = child;
-                    break;
-                }
+                // set up new node as folder node
+                newnode->type = FOLDERNODE;
+                newnode->nodehandle = 0; // why not UNDEF?
 
-                // there's an error if only one of the folder's {name, id} is correct and the other is wrong.
-                e = API_EINCOMPLETE;
-                break;
+                // generate fresh random key for this folder node
+                byte buf[FOLDERNODEKEYLENGTH];
+                client->rng.genblock(buf, FOLDERNODEKEYLENGTH);
+                newnode->nodekey.assign((char*)buf, FOLDERNODEKEYLENGTH);
+                SymmCipher key;
+                key.setkey(buf);
+
+                // generate fresh attribute object with the folder name
+                AttrMap attrs;
+                fsAccess->normalize(&deviceName);
+                attrs.map['n'] = deviceName;
+                nameid attrId = AttrMap::string2nameid("dev-id"); // "device-id" would be too long
+                attrs.map[attrId] = deviceId;
+
+                // JSON-encode object and encrypt attribute string
+                string attrstring;
+                attrs.getjson(&attrstring);
+                newnode->attrstring.reset(new string);
+                client->makeattr(&key, newnode->attrstring, attrstring.c_str());
+
+                // add the newly generated folder node
+                client->putnodes(h, move(newnodes));
+
+                // Get the new node & handle.
+                // HOW ? How can it wait until the remote folder was created?
+
+                // Update deviceNameNode somehow
             }
 
-            if (e != API_OK) { break; }
+            //
+            // create backupName remote dir
```

Comment:
There is a way to create multiple nodes with a single putnodes, as long as they belong to the same subtree (which is the case). It may be easier and more efficient, since you won't need to wait for the creation of the `DEVICE_NAME` folder in order to create the backup folder. Both would be done at the same time, with a single `p` command.
You will need to investigate the code in order to learn how to do it, I think a good bet is to search for usages of `NewNodes`, were more than one is added to the array. I think that's the way. If you don't find the way, ping the team on the channel.